slimta/envelope.py

```diff
@@ -24,19 +24,32 @@
 
 """
 
-from __future__ import absolute_import
+from __future__ import absolute_import, unicode_literals
 
 import re
 import copy
-import cStringIO
 from email.message import Message
 from email.generator import Generator
-from email.parser import Parser, FeedParser
+from email.parser import Parser
+
+import six
+from six.moves import cStringIO
+from six import BytesIO
+
+if six.PY2:
+    from email.generator import Generator as BytesGenerator
+
+else:
+    from email.parser import BytesParser
+    from email.generator import BytesGenerator
+
+from slimta.util.typecheck import check_argtype
+
 
 __all__ = ['Envelope']
 
-_HEADER_BOUNDARY = re.compile(r'\r?\n\s*?\n')
-_LINE_BREAK = re.compile(r'\r?\n')
+_HEADER_BOUNDARY = re.compile(b'\r?\n\s*?\n')
+_LINE_BREAK = re.compile('\r?\n')
```

Comment:
I'd like to leave the `r` prefix on all regex strings. For bytestring regexes, the prefix should be `br`.